data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/ownership/ConfigBucketOwnerProviderFactory.java

```diff
@@ -39,4 +44,15 @@ private String extractQueueAccountId(final S3SourceConfig s3SourceConfig) {
         }
         return accountId;
     }
+
+    private String extractStsRoleArnAccountId(final S3SourceConfig s3SourceConfig) {
+        final String stsRoleArn = s3SourceConfig.getAwsAuthenticationOptions().getAwsStsRoleArn();
+        final String accountId;
+        try {
+            accountId = StsArnRole.parse(stsRoleArn).getAccountId();
+        } catch (final MalformedURLException e) {
```

Comment:
StsArnRole.parse throws IllegalArgumentException. I don't see it throwing MalformedURLException.